File: specification/specs2/x_pack/graph/explore/GraphExploreRequest.ts
@rest_spec_name("xpack.graph.explore")
class GraphExploreRequest extends RequestBase {
	query: QueryContainer;
	vertices: GraphVertexDefinition[];
	connections: Hop;
	controls: GraphExploreControls;
	@request_parameter()
	routing: Routing;
	@request_parameter()
	timeout: Time;
}

File: output/csharp/XPack/Graph/Explore/GraphExploreRequest.cs
using Nest.CommonAbstractions;
using Nest.Internal;
using Nest.CommonOptions;
using Nest.XPack;
using Nest.QueryDsl;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class GraphExploreRequest : RequestBase {
		
		[DataMember(Name="routing")]
		public Routing Routing { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="connections")]
		public Hop Connections { get; set; }


		[DataMember(Name="controls")]
		public GraphExploreControls Controls { get; set; }


		[DataMember(Name="query")]
		public QueryContainer Query { get; set; }


		[DataMember(Name="vertices")]
		public List<GraphVertexDefinition> Vertices { get; set; }

	}
}
